tools/ncf.py

```diff
@@ -451,9 +465,16 @@ def generate_technique_content(technique_metadata):
 
   # Handle method calls
   for method_call in technique["method_calls"]:
+    method_name = method_call["method_name"]
     regex = re.compile(r'(?<!\\)"', flags=re.UNICODE )
     # Treat each argument of the method_call
     if 'args' in method_call:
+      for index, arg in enumerate(method_call['args']):
+        parameter = methods[method_name]["parameter"][index]
+        arg_constraint = parameter.get("constraints", {})
+        check = ncf_constraints.check_parameter(arg, arg_constraint)
```

Comment:
So we discussed in a meeting about this issue with @jooooooon @peckpeck 
In most case, when a variable is involved, checking contraint of it is undecidable and that it can only be checked at runtime (which is not the scope of this fix!) ( impossible for regex constraints, min length, at least very complicated for 'select' )

We decided that: when a variable is detected
* Skip constraint check that cannot be done
* Add an info log that constraint on that value could not be check
* Do all check that can be done (max_length, trailing/leading whitespaces)

@jooooooon @peckpeck , Am i missing something ?

@ncharles (as you reacted on this i ask you the question) and anyone who read that PR: What do you think of it ?